cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
I'm sorry it wasn't clear--let me provide more detail. If it's not clear enough, I'm happy to talk outside of a github issue (phone, hangout, Slack) if that helps.

When the Photon Controller kube-up implementation runs, it creates the VMs in some network. In my case, it's the 10.146.0.0/16 network. The VMs use DHCP to get IP addresses and the datacenter's DNS server, also in the 10.146.0.0/16 network. 

When the pods are created, they're in a private network. By default, we use 10.244.0.0/16, though it's configurable. The master and nodes have been configured with appropriate IP routes, and we're running kube-proxy (which configures IP tables) so communication between nodes, pods, and services work fine. As far as I understand, we meet the requirements listed at http://kubernetes.io/docs/admin/networking/ . That is, pod-to-pod, pod-to-service, and external-to-service network communications all function correctly. What we don't have (and I don't see in the networking model) is pod to external IP addresses. Perhaps it's a misunderstanding on my part?

Therefore when SkyDNS runs in a pod in our setup, it cannot contact the DNS server in the datacenter network (10.146.0.0/16). Enabling host networking allows it to contact the DNS server and everything works as expected.

What I found was that when SkyDNS is configured to inherit DNS but it can't contact the external DNS server, responses are very slow, even if it's only resolving something like kubernetes.default. Once communication with the external DNS works, responses are consistently fast. 

Does this help?